codice/ddf
# DDF-04269 Transformer blacklist for metacard and query response transformer export actions

#### What does this PR do?
- Adds a metacard and query response transformer blacklist so that specific transformers can be filtered out. 
- Removes the old way of doing this by setting a service property `export-resultset`

#### Who is reviewing it? 

@andrewzimmer 
@Bdthomson 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@brjeter
@emmberk

#### How should this be tested?
- Modify the metacard and query response transformer blacklists
- Verify that these export actions do not show up as export options for results and result sets. 

#### Any background context you want to provide?
- None

#### Screenshots:
Sample configuration
<img width="617" alt="screen shot 2019-02-07 at 11 19 54 am" src="https://user-images.githubusercontent.com/7646577/52433706-e5e10300-2aca-11e9-9744-1a6df3a61c48.png">
Result Export List
<img width="287" alt="screen shot 2019-02-07 at 11 20 37 am" src="https://user-images.githubusercontent.com/7646577/52433733-f4c7b580-2aca-11e9-8a04-6727433ff8b2.png">
Result Set Export List
<img width="1360" alt="screen shot 2019-02-07 at 11 21 08 am" src="https://user-images.githubusercontent.com/7646577/52433747-fc875a00-2aca-11e9-9e2a-ea54028412de.png">
Empty Blacklist Export Options
<img width="285" alt="screen shot 2019-02-07 at 11 20 57 am" src="https://user-images.githubusercontent.com/7646577/52433765-0741ef00-2acb-11e9-867a-870aaf5c4f5d.png">
<img width="1376" alt="screen shot 2019-02-07 at 11 22 56 am" src="https://user-images.githubusercontent.com/7646577/52433771-0a3cdf80-2acb-11e9-9418-7df22b5f3b3f.png">
Documentation
<img width="985" alt="screen shot 2019-02-07 at 12 17 30 pm" src="https://user-images.githubusercontent.com/7646577/52436783-6a834f80-2ad2-11e9-8ff7-2dbfb004624b.png">
<img width="993" alt="screen shot 2019-02-07 at 12 17 39 pm" src="https://user-images.githubusercontent.com/7646577/52436785-6ce5a980-2ad2-11e9-8b67-b4083ade5a58.png">

#### What are the relevant tickets?
Fixes: #4269

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
